packages/http-client/src/http-client.ts

```diff
@@ -54,14 +53,10 @@ const logger = createLogger({
 export async function buildHttpRequest(
   destination: DestinationOrFetchOptions
 ): Promise<DestinationHttpRequestConfig> {
-  const resolvedDestination = await resolveDestination(destination);
-  if (!resolvedDestination) {
-    throw Error(
-      `Failed to resolve the destination '${toDestinationNameUrl(
-        destination
-      )}'.`
-    );
-  }
+  const resolvedDestination = await resolveDestinationWithType(
```

Comment:
[q] This looks like a breaking or at least functional change if instead of a exception the logic is continued. Also why is the `buildHeaders` not complaining about the null?

Reply:
ah I just saw that the resolveDestinationWithType does the check so I guess it is fine.